app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditViewModel.kt

```diff
@@ -27,6 +25,9 @@ class ProfileEditViewModel @Inject constructor(
 
   lateinit var profileName: String
 
+  private val _activityTitle = MutableLiveData<String>()
```

Comment:
As per I understand from the issue and [this](https://github.com/oppia/oppia-android/blob/4c345db0f51c8a40870133113dd8ba7533a06d86/app/src/main/java/org/oppia/app/settings/profile/ProfileEditViewModel.kt) committed file, we have to work on `Switch`. 
@miaboloix Could you please correct me here?

Reply:
@anandwana001 you might need to approach @miaboloix directly w.r.t. this.

Reply:
@anandwana001 is correct, though there are actually two issues with this view model:
1. It relies on a view directly (the ``Switch``--view models should not depend on views)
2. It relies on Activity, but its lifecycle is managed by AndroidX's ViewModelProvider which means it will leak references to the previous activity during configuration changes

#1633 is meant to fix (1), but if possible we should also fix (2) while here. The way we've been fixing (2) everywhere else is just by injecting the view model directly into the presenter rather than using ViewModelProvider.

@alokbharti you'll need to figure out a way to make (1) work--we don't want to depend on ``Switch`` which may require introducing a callback interface or a live data from this view model.

Reply:
Hi @BenHenning , thanks for the suggestions. I'll fix those issues one by one for this viewmodel. For now, as suggested I'm doing the changes for the switch part.

Reply:
Thanks. While here, can you also do (2)? All that's needed is to switch the presenter from using ModelViewProvider to injecting the view model directly. There aren't straightforward tests that we can add for this, so we'll rely on existing tests to pass.